### PR TITLE
Adding lombok support to the model.Changing the Icon.

### PR DESCRIPTION
**Pull request by abdorah**
HI! I just tried to change the use of crud repository class and use instead Jpa class. I added lombok data to make it easier to the user of this extension to add only his or her attributes to the model without bothering to add constructors, getters and setters.I hope this would be somehow helpful!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "displayName": "Spring Boot Snippets",
     "description": "Useful snippets for Spring Boot projects.",
     "version": "1.0.1",
-    "icon": "images/spring-boot.png",
+    "icon": "images/spring-snippets.png",
```
Comment (abdorah) [inline package.json]: changing the icon.

File: snippets/java.json [hunk 2]
```diff
@@ -6,12 +6,11 @@
...
-            "import org.springframework.data.rest.core.annotation.RepositoryRestResource;",
-            "",
-            "//Remove @RepositoryRestResource below to disable auto REST api:",
-            "@RepositoryRestResource",
-            "public interface ${2:repositoryName} extends CrudRepository<${3:entityName}, ${4:entityIdType}>{}"
+            "import org.springframework.data.jpa.repository.JpaRepository;",
+            "import org.springframework.stereotype.Repository;",
+
+            "@Repository",
+            "public interface ${2:repositoryName} extends JpaRepository<${3:entityName}, ${4:entityIdType}>{}"
```
Comment (abdorah) [inline snippets/java.json]: Change the CrudRepository with JpaRepository

File: snippets/java.json [hunk 3]
```diff
@@ -26,24 +25,15 @@
             "import javax.persistence.GeneratedValue;",
             "import javax.persistence.GenerationType;",
             "import javax.persistence.Id;",
+            "import lombok.Data;",
             "",
+            "@Data",
```
Comment (abdorah) [inline snippets/java.json]: Adding lombok data. To use this you should add this dependency,e.g. in pom.xml file:
		```<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<optional>true</optional>
		</dependency>```

File: snippets/java.json [hunk 4]
```diff
@@ -26,24 +25,15 @@
...
-            "    public ${2:entityClassName}() {",
-            "    }",
-            "",
-            "    public ${3:entityIdType} getId() {",
-            "        return id;",
-            "    }",
-            "",
-            "    public void setId(${3:entityIdType} id) {",
-            "        this.id = id;",
-            "    }",
```
Comment (abdorah) [inline snippets/java.json]: No more need to add a constructor, getters and setters.

File: snippets/java.json (hunk 3 above)
Comment (DeveloperSoapbox) [reply snippets/java.json:30]: Hi Adborah!  Thanks for the feedback.  This is a reasonable request, but I would prefer to keep a "vanilla java" approach for those who do not use Lombok.  I personally like Lombok though so may add a new lombok specific snippet with your changes.

**Comment by DeveloperSoapbox** [comment]
Hi Adborah! Thanks for the feedback. This is a reasonable request.  I would prefer to keep a "vanilla java" approach for those who do not use Lombok, but I personally like Lombok too.  So I may add a new lombok specific snippet with your changes, and just have both options available.

**Comment by abdorah** [comment]
HI @DeveloperSoapbox ! Thank you for reviewing my changes I am glad you like it! Yes that would be wonderful if we get an other version of it with Lombok. I hope also you liked the image I add (you can use if you like it). I will be more than happy to help if you were up to add any new changes.

**Comment by DeveloperSoapbox** [comment]
@abdorah , since these will be additional snippets instead of replacements, would you mind using the code into new snippets in the java.json file?  

For the snippet prefix naming conventions, please use:
**spring-jparepo**: for the JPA repo changes
**spring-lombok-entity**: for the Lombok entity changes 

While I could simply create the new versions myself and copy and paste your code, I would like to simply merge the pull so that you can be recognized for these changes :)

File: package.json (hunk 1 above)
Comment (DeveloperSoapbox) [reply package.json]: I do like the image, but for my VS extensions I prefer not to use official logos to avoid infringing on trademarks.

**Comment by abdorah** [comment]
@DeveloperSoapbox  Yes thank you!  I will be happy to do it!

File: README.md [hunk 5]
```diff
@@ -16,7 +16,9 @@ Alternatively, press `Ctrl`+`Space` (Windows, Linux) or `Cmd`+`Space` (macOS) to
 | ---------------------------- | -------------------------------------------------------------------- |
 | `spring-crudctrl`            | Spring controller with CRUD actions implemented                      |
 | `spring-crudrepo`            | Spring Data JPA CRUD repository with REST api annotation             |
+| `spring-jparepo`            | Spring Data JPA repository with REST api annotation             |
 | `spring-entity`              | JPA entity with ID field                                             |
+| `spring-lombok-entity`              | JPA entity with ID field and Lombok `@Data` annotation                                            |
```
Comment (abdorah) [inline README.md:21]: Snippets description To either use Lombok or not and JPA repository or CRUD repository.

File: snippets/java.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
     "Spring JPA Rest Repository": {
         "prefix": [
-            "spring-crudrepo"
+            "spring-jparepo"
```
Comment (abdorah) [inline snippets/java.json]: Changing prefix to `spring-jparepo`

File: snippets/java.json [hunk 7]
```diff
@@ -12,9 +12,25 @@
...
+            "import org.springframework.data.rest.core.annotation.RepositoryRestResource;",
+            "",
+            "//Remove @RepositoryRestResource below to disable auto REST api:",
+            "@RepositoryRestResource",
+            "public interface ${2:repositoryName} extends CrudRepository<${3:entityName}, ${4:entityIdType}>{}"
+        ],
         "description": "JPA CRUD repository with REST api annotation"
     },
-    "Spring JPA Entity": {
+    "Spring JPA Entity with Lombok": {
```
Comment (abdorah) [inline snippets/java.json]: Changing name to specify in which snippets we used Lombok.

File: snippets/java.json [hunk 8]
```diff
@@ -36,8 +53,41 @@
             "    private ${3:entityIdType} id;",
             "}"
         ],
-        "description": "JPA entity with ID field"
+        "description": "JPA entity with ID field and Lombok data annotation"
```
Comment (abdorah) [inline snippets/java.json]: Changing description to specify in which snippets we used Lombok.

**Review by abdorah — COMMENTED** [review]
HI @DeveloperSoapbox ! You can now check my changes and in case of no problem accept and merge my pull request. Please remember to change the version and thank you!

File: package.json (hunk 1 above)
Comment (abdorah) [reply package.json]: Oh! I understand! I really don't have any ideas of what the legit logo would be. But if you have any ideas about the logo I will be happy to help.

**Review by DeveloperSoapbox — APPROVED** [review]
Great work @abdorah .  The only thing I noticed is that the prefix for the lombok entity snippet needs to be updated, but I can take care of that really quick.

**Review by DeveloperSoapbox — CHANGES_REQUESTED** [review]
Sorry @abdorah , I had not noticed the logo image change was still applied.  Before I accept could you please revert the deleted image change and the image reference in the package.json file?  

In the end, the merge request should have change requests only to README.md and java.json file.

Thank you!

**Comment by abdorah** [comment]
Oh I forgot to change the image! I think that now every thing should work very well! Thank you!

**Review by DeveloperSoapbox — APPROVED** [review]
Thanks @abdorah for the super quick turnaround.

**Comment by abdorah** [comment]
Thank you too @DeveloperSoapbox ! I am very happy to help!

**Pull request merged**